### PR TITLE
Deployment test fixes

### PR DESCRIPTION
**Pull request by pperiyasamy**
This includes fixes for issues in Dockerfile, Makefile, authenticator crd and rbac that are found during while deploying eapol operator and authenticator daemonset.

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -5,6 +5,12 @@ generatorOptions:
...
-  files:
+- files:
   - controller_manager_config.yaml
+  name: manager-config
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+images:
+- name: controller
+  newName: controller
+  newTag: latest
```
Comment (lack) [inline config/manager/kustomization.yaml:18]: Can you explain this one?  It seems odd and out-of-order.

Comment (pperiyasamy) [reply config/manager/kustomization.yaml:18]: @lack This was auto generated by this kustomize [command](https://github.com/openshift-kni/eapol-operator/blob/main/Makefile#L153). I just added it to make sure `git diff` to be intact even after running `make deploy`.

Comment (lack) [reply config/manager/kustomization.yaml:18]: Can we reorder it a bit more and put the `apiVersion` and `kind` at the top?  The rest of the reordering is fine.

Comment (pperiyasamy) [reply config/manager/kustomization.yaml:18]: oh! yes. done.

**Review by lack — APPROVED** [review]

**Pull request merged**